scripts/terra_service_banner.py

```diff
@@ -0,0 +1,98 @@
+import argparse
+from google.cloud import storage as gcs
+
+
+def convert_json_to_string(json, env):
+    """Convert json file to string if supplied with custom banner json file."""
+
+    # open json file and read contents to a string
+    with open(json, "r") as j:
+        custom_banner_string = j.read()
+
+    print(f'Starting custom banner upload to {env} Terra.')
+    update_service_banner(custom_banner_string, env)
+
+
+def update_service_banner(json, env):
+    """Push json to bucket in selected production environment."""
+
+    # create storage Client and assign destination bucket
+    storage_client = gcs.Client()
+
+    # set bucket and suitable_group based on env
+    if env == "prod":
+        bucket = storage_client.get_bucket("firecloud-alerts")
+        suitable_group = "fc-comms@firecloud.org"
+    else:
+        bucket = storage_client.get_bucket(f"firecloud-alerts-{env}")
+        suitable_group = (f"fc-comms@{env}.test.firecloud.org")
+
+    # define required filename (alerts.json) and upload json string to gcs
+    blob = bucket.blob("alerts.json")
+    blob.upload_from_string(json)
+
+    print(f'Setting permissions and security on banner json object in GCS location.')
+    # set metadata on json object (gsutil -m setmeta -r -h "Cache-Control:private, max-age=0, no-cache")
+    blob.cache_control = "private, max-age=0, no-cache"
+    blob.patch()
+
+    # set and save READ access for AllUsers on json object (gsutil ach ch -u AllUsers:R)
+    blob.acl.all().grant_read()
+    blob.acl.save()
+
+    # set and save OWNER access for suitable_group on json object (gsutil ach ch -g suitable_group:O)
+    blob.acl.group(suitable_group).grant_owner()
+    blob.acl.save()
+
+    print("Banner action complete.")
+
+
+def post_banner(env):
+    """Create json string for upload to GCS location to post/publish banner."""
+
+    # template json text for banner publication
+    banner_text = """[
+                        {
+                        "title":"Service Incident",
+                        "message":"We are currently investigating an issue impacting the platform. Information about this incident will be made available here.",
+                        "link":"https://support.terra.bio/hc/en-us/sections/360003692231-Service-Notifications"
+                        }
+                    ]"""
+
+    # push json string to bucket - post banner
+    print(f'Starting template banner upload to {env} Terra.')
+    update_service_banner(banner_text, env)
+
+
+def delete_banner(env):
```

Comment:
- is it the case that only one banner can ever be up on Terra? 
if not: 
- does this function remove all banners? would we ever need to remove one but keep another?

Reply:
For the sole purpose of service incidents, this script acts on only a single object (specifically named "alerts.json") in a specific bucket. It only modifies the contents of that .json and Terra UI reads from that object. If there are contents, it posts a banner, and if its empty, it shows no banner (aka delete banner). There are no other banners on Terra that Frontline or anyone using this script could activate/modify etc. The other things that might pop up are errors that appear in the same shape/form as a banner but are not banners. The only thing someone can take down (delete) is the same banner that is put up via that "alerts.json" object.

Reply:
got it, that makes sense, thanks! maybe worth including that explanation in the readme, just for someone new to the whole thing?

and to me that makes this function more like `clear_all_service_banners` but that's a personal preference!

Reply:
totally! thats great feedback to have for someone that is new to the process! ill make the function different